src/factory/client/clients.ts

```diff
@@ -136,17 +133,25 @@ export function createEVMClient(
   const swapProvider = new EvmSwapProvider(swapOptions);
 
   let walletProvider;
-  if (accountType === AccountType.EthereumLedger || accountType === AccountType.RskLedger) {
+  if (accountInfo.type === AccountType.EthereumLedger || accountInfo.type === AccountType.RskLedger) {
+    let addressCache;
+    if (accountInfo && accountInfo.publicKey && accountInfo.address) {
+      addressCache = new Address({
+        publicKey: accountInfo?.publicKey,
+        address: accountInfo.address
+      });
+    }
     walletProvider = new EvmLedgerProvider(
       {
         network: ethereumNetwork,
-        derivationPath,
+        derivationPath: accountInfo.derivationPath,
+        addressCache,
         transportCreator: ledgerTransportCreator,
```

Comment:
We can't asume that all clients will be able to support WebHID transport. 
This should be an option in `WalletOptions` similar to how ledger providers were handled before

Reply:
in this case seems like wallet core can handle using it and it's implemented inside the '@chainify/hw-ledger' in `WebHidTransportCreator` I will add it then to the wallet and send inside `WalletOptions` like you said

Reply:
fixed